announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java

```diff
@@ -227,59 +230,51 @@ public String getEntityPrefix() {
 		//get the announcements for each channel
 		List<Message> announcements = new ArrayList<Message>();
 		
-		boolean enableReorder = ComponentManager.get(ServerConfigurationService.class).getBoolean(AnnouncementAction.SAK_PROP_ANNC_REORDER, AnnouncementAction.SAK_PROP_ANNC_REORDER_DEFAULT);
+		boolean enableReorder = serverConfigurationService.getBoolean(AnnouncementAction.SAK_PROP_ANNC_REORDER, AnnouncementAction.SAK_PROP_ANNC_REORDER_DEFAULT);
 		String sortCurrentOrder = AnnouncementAction.SORT_DATE;
 		if (enableReorder){
 			sortCurrentOrder = AnnouncementAction.SORT_MESSAGE_ORDER;
 		}
-		ViewableFilter msgFilter = AnnouncementAction.SORT_MESSAGE_ORDER.equals(sortCurrentOrder) ? null:new ViewableFilter(null, t, numberOfAnnouncements);
-		
-		//for each channel
-		for(String channel: channels) {
-			try {
-				announcements.addAll(announcementService.getMessages(channel, msgFilter, announcementSortAsc, false));
-			} catch (PermissionException | IdUnusedException | NullPointerException ex) {
-				//user may not have access to view the channel but get all public messages in this channel
-				AnnouncementChannel announcementChannel = (AnnouncementChannel)announcementService.getChannelPublic(channel);
-				if(announcementChannel != null){
-					List<Message> publicMessages = announcementChannel.getMessagesPublic(null, true);
-					for(Message message : publicMessages){
-						//Add message only if it is within the time range
-						if(isMessageWithinPastNDays(message, numberOfDaysInThePast) && announcementService.isMessageViewable((AnnouncementMessage) message)){
-							announcements.add(message);
-						}
-					}
-				}
-			}
-		}
-		
-		if(log.isDebugEnabled()) {
-			log.debug("announcements.size(): {}", announcements.size());
-		}
+		ViewableFilter msgFilter = AnnouncementAction.SORT_MESSAGE_ORDER.equals(sortCurrentOrder) ? null : new ViewableFilter(null, t, numberOfAnnouncements);
 		
 		if (AnnouncementAction.SORT_MESSAGE_ORDER.equals(sortCurrentOrder)) {
 			try {
 				List<AnnouncementWrapper> messageList = new ArrayList<>();
-				AnnouncementChannel defaultChannel = (AnnouncementChannel) announcementService.getChannel("/announcement/channel/" + siteId + "/main");
+				final AnnouncementChannel defaultChannel = (AnnouncementChannel) announcementService.getChannel("/announcement/channel/" + siteId + "/main");
 				for (Message msg : announcements) {
```

Comment:
the **announcements** list is always empty here because the else section below used to be above the sort and always populated the list first.

Reply:
i don't follow. announcements is always empty on line 244?

Reply:
yes on line 244
{code}
for (Message msg : announcements) {
{code}
due to Earle's change, when the code gets to line 244, announcements haven't been added to the announcements list yet.  Because the block that used to set it got moved into an else block.

Reply:
okay can you make the change to fix it ?

Reply:
ok, will do

Reply:
committed latest fix

but i'm not totally sure about these lines
```
announcements.removeAll(announcements);
announcements.addAll(messageList);
```